### PR TITLE
docs: Fix typos on certain pages of Keep Docs Site

### LINKED ISSUE #2294: [📃 Docs]: Fix typos on certain pages of Keep Docs Site
In the pages listed below, there are a few typos that need to be fixed. Find the page links and the screenshots attached for reference.

#### [One Until Resolved Page](https://docs.keephq.dev/workflows/throttles/one-until-resolved)

Screenshot 2024-10-24 212031 (1)

#### [Foreach Page](https://docs.keephq.dev/workflows/syntax/foreach-syntax)

Screenshot 2024-10-24 212407 (1)

#### [Working with context Page](https://docs.keephq.dev/workflows/syntax/context-syntax)

Screenshot 2024-10-24 212817 (1)

#### [Basic Syntax Page](https://docs.keephq.dev/workflows/syntax/basic-syntax)

Screenshot 2024-10-24 213129 (1)
Screenshot 2024-10-24 213107 (1)
Screenshot 2024-10-24 213050 (1)

#### [State Page](https://docs.keephq.dev/workflows/state)

Screenshot 2024-10-24 213611 (1)
Screenshot 2024-10-24 213542 (1)

### PR DESCRIPTION
**Pull request by Rutam21**
Fixes #2294 #2297 

## 📑 Description

This PR fixes typos on the following pages of the Keep Docs site.

-  **[One Until Resolved Page](https://docs.keephq.dev/workflows/throttles/one-until-resolved)**
-  **[Foreach Page](https://docs.keephq.dev/workflows/syntax/foreach-syntax)**
-  **[Working with context Page](https://docs.keephq.dev/workflows/syntax/context-syntax)**
-  **[Basic Syntax Page](https://docs.keephq.dev/workflows/syntax/basic-syntax)**
-  **[State Page](https://docs.keephq.dev/workflows/state)**
-  **[Auth0 Page](https://docs.keephq.dev/providers/documentation/auth0-provider)**
-  **[Adding a new Provider Page](https://docs.keephq.dev/providers/adding-a-new-provider)**
-  **[Key concepts Page](https://docs.keephq.dev/overview/keyconcepts)**
-  **[Introduction Page](https://docs.keephq.dev/overview/introduction)**
-  **[Examples Page](https://docs.keephq.dev/overview/examples)**

## ✅ Checks
- [x] This PR follows the Keep Contributing Guidelines.

### CODE REVIEW

**Comment by Rutam21** [comment]
@talboren Please review and suggest changes, if any. Thanks.

**Comment by Rutam21** [comment]
@talboren Please review and suggest changes, if any. Thanks.

cc: @shahargl

**Comment by Rutam21** [comment]
@talboren I have merged the PRs into this one. Please review and merge. Thanks.

File: docs/workflows/syntax/basic-syntax.mdx [hunk 1]
```diff
@@ -126,5 +126,5 @@ on-failure:
       config: " {{ providers.slack-demo }} "
 ```
 
-On-failure is actually a condtionless `Action` used to notify in case the alert failed with an exception.
+On-failure is actually an unconditional `Action` used to notify in case the alert failed with an exception.
```
Comment (talboren) [inline docs/workflows/syntax/basic-syntax.mdx:129]: The meaning was an action with no condition. Unconditional sounds weird.

**Review by talboren — CHANGES_REQUESTED** [review]
1 comment

File: docs/workflows/syntax/basic-syntax.mdx (hunk 1 above)
Comment (Rutam21) [reply docs/workflows/syntax/basic-syntax.mdx:129]: @talboren Yeah, I got that meaning. But `conditionless` is not a correct word. If `unconditional` sounds weird, then we can simply keep it as `without any condition` or `with no condition`. We can also use `an unconditioned Action`. Please let me know what you think about it.

Comment (talboren) [reply docs/workflows/syntax/basic-syntax.mdx:129]: with no condition sounds best.

Comment (Rutam21) [reply docs/workflows/syntax/basic-syntax.mdx]: Thanks. Just so you know – all suggestions are applied.

**Review by talboren — APPROVED** [review]
LGTM!

**Pull request merged**